samples/drivers/flash_shell/src/main.c

```diff
@@ -372,11 +372,11 @@ static int do_write_unaligned(const struct shell *shell, off_t offset, uint8_t *
 	k_free(before_data);
 	k_free(after_data);
 
-	ret = flash_write_protection_set(flash_device, true);
-	if (ret) {
+	int ret1 = flash_write_protection_set(flash_device, true);
+	if (ret1) {
 		PR_ERROR(shell, "Failed to enable flash protection (err: %d)."
-				"\n", ret);
-		return ret;
+				"\n", ret1);
+		return ret1;
```

Comment:
it seems that the error is meaningless, I mean, the check is just for something != 0. So you could re-use the variable `ret` and return a negative value later, another thing is that we don't declare variables ad-hoc. Though the fix is correct and avoid possible return 0 in case of error.